pkg/cmd/taskrun/list_test.go

```diff
@@ -0,0 +1,169 @@
+				"NAME    STARTED      DURATION   STATUS      ",
+				"tr1-1   1 hour ago   1 minute   Succeeded   ",
+				"",
+			},
+		},
+		{
+			name:    "all in namespace",
+			command: command(trs, now),
+			args:    []string{"list", "-n", "foo"},
+			expected: []string{
+				"NAME    STARTED      DURATION   STATUS               ",
+				"tr1-1   1 hour ago   1 minute   Succeeded            ",
+				"tr2-1   1 hour ago   ---        Succeeded(Running)   ",
+				"tr2-2   1 hour ago   1 minute   Failed               ",
+				"",
+			},
+		},
+		{
+			name:    "print by template",
+			command: command(trs, now),
+			args:    []string{"list", "-n", "foo", "-o", "jsonpath={range .items[*]}{.metadata.name}{\"\\n\"}{end}"},
+			expected: []string{
+				"tr1-1",
+				"tr2-1",
+				"tr2-2",
+				"",
+			},
+		},
+		{
+			name:     "empty list",
+			command:  command(trs, now),
+			args:     []string{"list", "-n", "random"},
+			expected: []string{msgNoTRsFound, ""},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := testutil.ExecuteCommand(test.command, test.args...)
+
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+			if d := cmp.Diff(strings.Join(test.expected, "\n"), got); d != "" {
+				t.Errorf("Unexpected output mismatch: \n%s\n", d)
+			}
+		})
+	}
+}
+
+func command(trs []*v1alpha1.TaskRun, now time.Time) *cobra.Command {
+	// fake clock advanced by 1 hour
+	clock := clockwork.NewFakeClockAt(now)
+	clock.Advance(time.Duration(60) * time.Minute)
+
+	cs, _ := test.SeedTestData(test.Data{TaskRuns: trs})
+
+	p := &testutil.TestParams{Client: cs.Pipeline, Clock: clock}
+
+	return Command(p)
+}
+
+func taskRuns(start time.Time) []*v1alpha1.TaskRun {
+	aMinute, _ := time.ParseDuration("1m")
+
+	trsData := []struct {
+		name       string
+		ns         string
+		task       string
+		status     corev1.ConditionStatus
+		reason     string
+		startTime  time.Time
+		finishTime time.Time
+	}{
+		{
+			name:       "tr1-1",
+			ns:         "foo",
+			task:       "bar",
+			status:     corev1.ConditionTrue,
+			reason:     resources.ReasonSucceeded,
+			startTime:  start,
+			finishTime: start.Add(aMinute),
+		},
+		{
+			name:      "tr2-1",
+			ns:        "foo",
+			task:      "random",
+			status:    corev1.ConditionTrue,
+			reason:    resources.ReasonRunning,
+			startTime: start,
+		},
+		{
+			name:       "tr2-2",
+			ns:         "foo",
+			task:       "random",
+			status:     corev1.ConditionFalse,
+			reason:     resources.ReasonFailed,
+			startTime:  start,
+			finishTime: start.Add(aMinute),
+		},
+	}
+
+	trs := []*v1alpha1.TaskRun{}
+	for _, data := range trsData {
+		tr := tb.TaskRun(data.name, data.ns,
+			tb.TaskRunLabel("tekton.dev/task", data.task),
+			tb.TaskRunStatus(
+				tb.Condition(apis.Condition{
+					Status: data.status,
+					Reason: data.reason,
+				}),
+				tb.TaskRunStartTime(data.startTime),
+			),
+		)
+
+		tr.Status.CompletionTime = &metav1.Time{Time: data.finishTime}
```

Comment:
IMHO this need to be  moved to   `testutils/tekton.go` as `TaskRunOp` .

Reply:
Same as above :stuck_out_tongue_closed_eyes: 
I should make it a function for sure though :sweat_smile: